homeassistant/components/trafikverket_weatherstation/sensor.py

```diff
@@ -197,29 +209,32 @@ class TrafikverketWeatherStation(SensorEntity):
 
     def __init__(
         self,
-        weather_api,
-        name,
-        sensor_station,
+        weather_api: TrafikverketWeather,
+        name: str,
+        sensor_station: str,
         description: TrafikverketSensorEntityDescription,
-    ):
+    ) -> None:
         """Initialize the sensor."""
         self.entity_description = description
         self._attr_name = f"{name} {description.name}"
+        self._attr_unique_id = f"trafikverket_{name}_{description.key}"
         self._station = sensor_station
         self._weather_api = weather_api
-        self._weather = None
+        self._weather: WeatherStationInfo | None = None
+        self._active: bool = False
+        self._measure_time: str = "1970-01-01T00:00:00"
```

Comment:
It the state attribute value is unknown our standard is to not include the item at all in the state attributes until we have a value.

Reply:
Doing that will cause automation to behave unexpectedly, e.g., when using triggers. As state triggers or conditions cannot differentiate between unexpected missing or not having a value. Therefore I still strongly believe of making it a `None` value is a better approach; to prevent automations breaking on missing attributes.

Reply:
Well our current standard is like this. Until we decide otherwise we should follow that.

Reply:
These are not documented standards. I also review it exactly the other way around for a long time. I will keep doing that until this has been decided on; as I strongly believe your suggestion is causing problems.

Reply:
Please don't do that. The majority standard is what I've described.

We should keep it as consistent as possible until we decide otherwise.

Reply:
> Please don't do that. The majority standard is what I've described.

My statement of the above still holds; I have been doing that for as long as I have been touching Home Assistant and will keep doing that. There is no policy on this and recently we (with @emontnemery) that again showed it causing problems.

I will not change my opinion on this simply because of these above comments. Unless a policy is created for this, I will not change my review behavior on this one.

Reply:
You're not making it easy for me 😃  and personally I am in favor off what @frenck is suggesting besides using them in templated, automations etc. they are also directly visible for the user knowing what information is provided/will be provided.
Now on this particular integration I don't think it will make much difference so will go with adding them once they are updated from the api.

Reply:
Architectural discussion: https://github.com/home-assistant/architecture/discussions/680